pandas/core/generic.py

```diff
@@ -2368,18 +2371,58 @@ def filter(self, items=None, like=None, regex=None, axis=None):
         regex : string (regular expression)
             Keep info axis with re.search(regex, col) == True
         axis : int or None
-            The axis to filter on. By default this is the info axis. The "info
-            axis" is the axis that is used when indexing with ``[]``. For
-            example, ``df = DataFrame({'a': [1, 2, 3, 4]]}); df['a']``. So,
-            the ``DataFrame`` columns are the info axis.
+            The axis to filter on.  By default this is the info axis. The "info
+            axis" is the axis that is used when indexing with ``[]``
+
+        Returns
+        -------
+        same type as input object with filtered info axis
+
+        Examples
+        --------
+        >>> df
+        one  two  three
+        mouse     1    2      3
+        rabbit    4    5      6
+
+        >>> # select columns by name
+        >>> df.filter(items=['one', 'three'])
+        one  three
+        mouse     1      3
+        rabbit    4      6
+
+        >>> # select columns by regular expression
+        >>> df.filter(regex='e$', axis=1)
+        one  three
+        mouse     1      3
+        rabbit    4      6
+
+        >>> # select rows containing 'bbi'
+        >>> df.filter(like='bbi', axis=0)
+        one  two  three
+        rabbit    4    5      6
+
+        See Also
+        --------
+        pandas.DataFrame.select
 
         Notes
         -----
-        Arguments are mutually exclusive, but this is not checked for
+        The ``items``, ``like``, and ``regex`` parameters should be
+        mutually exclusive, but this is not checked.
 
+        ``axis`` defaults to the info axis that is used when indexing
+        with ``[]``.
         """
         import re
 
+        args = locals().copy()
+        nkw = sum(map(lambda x: operator.getitem(args,x)!=None, ['items', 'like', 'regex']))
```

Comment:
This seems a bit complicated for what we want to achieve (the locals is certainly not needed I think). 
Using a similar construct: `sum(map(lambda x: x is not None, [item, like, regex]))` (but using a list comprehension is maybe easier to read: `sum([x is not None for x in [item, like, regex]])`)